bdns/dns_test.go

```diff
@@ -422,13 +447,12 @@ func TestRetry(t *testing.T) {
 		if err == errTooManyRequests {
 			t.Errorf("#%d, sent more requests than the test case handles", i)
 		}
-		expectedErr := tc.te.errs[tc.expected-1]
-		if err != expectedErr {
+		expectedErr := tc.expected
+		if (expectedErr == nil && err != nil) ||
+			(expectedErr != nil && err == nil) ||
+			(expectedErr != nil && expectedErr.Error() != err.Error()) {
 			t.Errorf("#%d, error, expected %v, got %v", i, expectedErr, err)
 		}
-		if tc.expected != tc.te.count {
```

Comment:
This seems to just be checking if the resolver made the correct number of attempts against the server, since you switched `expected` to a `error` you can't check against `len(te.errs)` so I guess you need another var that counts the number of expected attempts?